group.py

```diff
@@ -418,8 +418,19 @@ def api_group_create(ctx, group_name, category, subcategory, description, data_c
     :param subcategory:         Subcategory of the group to create
     :param description:         Description of the group to create
     :param data_classification: Data classification of the group to create
+
+    :returns: Dict with API status result
     """
-    ctx.uuGroupAdd(group_name, category, subcategory, description, data_classification, '', '')
+    try:
+        response = ctx.uuGroupAdd(group_name, category, subcategory, description, data_classification, '', '')['arguments']
+        status = response[5]
+        message = response[6]
+        if status == '0':
+            return api.Result.ok()
+        else:
+            return api.Error('-1', message)
```

Comment:
Instead of -1 we can give the error a name to influence the http status returned by the portal (https://github.com/UtrechtUniversity/yoda-portal/blob/26b233cc613ba0227de482f4ac16c23ab25312a3/api.py#L30).

When an policy error occurs, we can use the name 'policy_error' (something other than error_internal and ok). This will result in http status 400. We want http status 400, because this was an invalid request from the client to the server.

Reply:
Changed status into 'error_internal' and 'policy_error' at the proper locations 👍